### PR TITLE
demos(Card): added horizontal split and details demos

### LINKED ISSUE #6121: Card: Horizontal Split Demo

### LINKED ISSUE #6122: Card: Details demo

### PR DESCRIPTION
**Pull request by tlabaj**
**What**: Closes #6122 and closes #6121

### CODE REVIEW

**Review by mcarrano — COMMENTED** [review]
Looks good to me. @mattnolting can you also take a look? The only question I had was why this URL on the right is not linked:

Screen Shot 2021-08-19 at 9 46 29 AM

It's that way in the core example, too. I wasn't sure if this was by intent or just an oversight.

File: packages/react-core/src/demos/CardDemos.md [hunk 1]
```diff
@@ -260,3 +253,163 @@ class CardGridDemo extends React.Component {
...
+          </Title>
+        </CardTitle>
+        <CardBody>
+          <DescriptionList>
+            <DescriptionListGroup>
+              <DescriptionListTerm>
+                Cluster API Address
+              </DescriptionListTerm>
+              <DescriptionListDescription>
+                https://api2.devcluster.openshift.com
```
Comment (kmcfaul) [inline packages/react-core/src/demos/CardDemos.md:372]: This feels a bit odd to see a link as plain text. Could we change this to a link, or change the description/descriptionListTerm to use less link-looking text?

File: packages/react-core/src/demos/CardDemos.md [hunk 2]
```diff
@@ -260,3 +253,163 @@ class CardGridDemo extends React.Component {
...
+  CardFooter,
+  Grid,
+  GridItem,
+  Button
+} from '@patternfly/react-core';
+
+CardHorizontalSplitDemo = () => {
+  return (
+    <Card id="card-demo-horizontal-split-example" isFlat>
+      <Grid md={6} hasGutter>
```
Comment (mcoker) [inline packages/react-core/src/demos/CardDemos.md:274]:
```suggestion
      <Grid md={6}>
```

File: packages/react-core/src/demos/CardDemos.md [hunk 3]
```diff
@@ -260,3 +253,163 @@ class CardGridDemo extends React.Component {
...
+        <GridItem style={{minHeight: '200px', backgroundPosition: 'center', backgroundSize: 'cover', backgroundImage: 'url(/assets/images/pfbg_992@2x.jpg)' }}/>
+        <GridItem>
+          <CardTitle>Headline</CardTitle>
```
Comment (mcoker) [inline packages/react-core/src/demos/CardDemos.md:277]: The `<Card*>` elements should be wrapped in `<Card>`. We're missing this on the core side, too - https://github.com/patternfly/patternfly/issues/4310

File: packages/react-core/src/demos/CardDemos.md [hunk 4]
```diff
@@ -260,3 +253,163 @@ class CardGridDemo extends React.Component {
...
+            <Button variant="tertiary">Call to action</Button>
+          </CardFooter>
+        </GridItem>
+      </Grid>
+    </Card>
+  );
+}
+```
+
+### Card details
```
Comment (mcoker) [inline packages/react-core/src/demos/CardDemos.md:291]: It's called "Details card" on the core side - @mcarrano how should this one be named?

**Review by mcoker — CHANGES_REQUESTED** [review]
Looks great! Just a few comments.

File: packages/react-core/src/demos/CardDemos.md (hunk 1 above)
Comment (mcoker) [reply packages/react-core/src/demos/CardDemos.md:372]: Yeah this seems like an oversight. I think we should link this text like the other card - @mcarrano do you agree?

Comment (tlabaj) [reply packages/react-core/src/demos/CardDemos.md:372]: Yup.  I agree. it looks odd. This is how the core implementation looks.  @mnolting can you see @mcarrano's question above
https://github.com/patternfly/patternfly-react/pull/6193#pullrequestreview-734011794

Comment (mcoker) [reply packages/react-core/src/demos/CardDemos.md:372]: It looks like neither of the URLs are linked in the original design - https://marvelapp.com/prototype/eb14d81/screen/70405126

File: packages/react-core/src/demos/CardDemos.md (hunk 3 above)
Comment (tlabaj) [reply packages/react-core/src/demos/CardDemos.md:277]: @mcoker I am confused about the structure.  I assumed the `<Card>` on line 273 was the wrapping  element here.

Comment (mcoker) [reply packages/react-core/src/demos/CardDemos.md:277]: Ah hah! I missed that, ignore my comment 🤦‍♂️

File: packages/react-core/src/demos/CardDemos.md (hunk 1 above)
Comment (mcarrano) [reply packages/react-core/src/demos/CardDemos.md:372]: Yes, Let's just make that a link for consistency.

Comment (tlabaj) [reply packages/react-core/src/demos/CardDemos.md:372]: @mcoker Do you want to open a core issue for the update.

**Review by CooperRedhat — APPROVED** [review]
LGTM

**Review by mcarrano — APPROVED** [review]
Looks good. Thanks @tlabaj !

**Review by dlabrecq — APPROVED** [review]

**Review by kmcfaul — APPROVED** [review]

**Review by mcoker — APPROVED** [review]
LGTM!

**Pull request merged**